alert-database/src/main/java/com/synopsys/integration/alert/database/api/StaticJobAccessor.java

```diff
@@ -207,15 +208,23 @@ public StaticJobAccessor(
 
         // TODO running project name pattern checks in java code, try to do this in SQL instead (Won't need to return DistributionJobEntity anymore if this happens
         return distributionJobEntities.stream()
-                   .filter(distributionJobEntity -> !(distributionJobEntity.getBlackDuckJobDetails().getFilterByProject() &&
-                                                          distributionJobEntity.getBlackDuckJobDetails().getProjectNamePattern() != null &&
-                                                          !Pattern.matches(distributionJobEntity.getBlackDuckJobDetails().getProjectNamePattern(), projectName) &&
-                                                          !distributionJobEntity.getBlackDuckJobDetails().getBlackDuckJobProjects().contains(projectName))
-                   )
+                   .filter(distributionJobEntity -> filterByProjects(distributionJobEntity, projectName))
                    .map(this::convertToFilteredDistributionJobResponseModel)
                    .collect(Collectors.toList());
     }
 
+    private boolean filterByProjects(DistributionJobEntity distributionJobEntity, String projectName) {
+        BlackDuckJobDetailsEntity blackDuckJobDetails = distributionJobEntity.getBlackDuckJobDetails();
+        String projectNamePattern = blackDuckJobDetails.getProjectNamePattern();
+        return !(blackDuckJobDetails.getFilterByProject() &&
+                     projectNamePattern != null &&
+                     !Pattern.matches(projectNamePattern, projectName) &&
+                     blackDuckJobDetails.getBlackDuckJobProjects()
+                         .stream()
+                         .map(BlackDuckJobProjectEntity::getProjectName)
+                         .noneMatch(projectName::equals));
```

Comment:
This is very confusing. You are negating the entire condition? Why not change the conditions to what you would expect them to be?
EX:
``` java
!blackDuckJobDetails.getFilterByProject() || 
(projectNamePattern != null && Pattern.matches(projectNamePattern, projectName)) ||
blackDuckJobDetails.getBlackDuckJobProjects()
                         .stream()
                         .map(BlackDuckJobProjectEntity::getProjectName)
                         .anyMatch(projectName::equals))
```